### PR TITLE
Reduce heights of navbar and footer

### PR DESCRIPTION
**Pull request by janakitti**

### CODE REVIEW

File: src/app/pages/signing-auth-redirect/signing-auth-redirect.component.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export class SigningAuthRedirectComponent implements OnInit {
 
   goToSigning() {
     // noinspection JSIgnoredPromiseFromCall
-    this.authService.login('/signing', { uid: this.uid, key: this.key });
+    this.authService.login('signing', { uid: this.uid, key: this.key });
```
Comment (ricetech) [inline src/app/pages/signing-auth-redirect/signing-auth-redirect.component.ts:16]: Incorrect usage of routerLink
```suggestion
    this.authService.login('/signing', { uid: this.uid, key: this.key });
```

File: src/app/pages/viewing/messages/messages.component.html [hunk 2]
```diff
@@ -8,36 +8,59 @@
...
+            </tr>
+          </table>
+        </div>
+        <div class="card-header-group">
+          <table class="header-table">
+            <tr>
+              <td>
+                <button class="btn card-button" type="button" data-toggle=""
+                        data-target="#bookmark">
+                  <img class="card-icon" src="../../../../assets/images/icons/bookmark.svg">
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.html:49]: Missing required alt attribute
```suggestion
                  <img class="card-icon" src="../../../../assets/images/icons/bookmark.svg" alt="Bookmark Message">
```

File: src/app/pages/viewing/messages/messages.component.html [hunk 3]
```diff
@@ -8,36 +8,59 @@
...
+                </button>
+                <button class="btn card-button" type="button" data-toggle="modal"
+                        data-target="#delete" (click)="setMessageToDelete(msg.id)">
+                  <img class="card-icon" src="../../../../assets/images/icons/trash.svg">
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.html:53]: Missing required alt attribute
```suggestion
                  <img class="card-icon" src="../../../../assets/images/icons/trash.svg" alt="Delete Message">
```

File: src/app/pages/viewing/responses-list/responses-list.component.html [hunk 4]
```diff
@@ -1,70 +1,32 @@
...
-        <!--suppress HtmlUnknownAttribute -->
-        <li class="nav-item mb-1" *ngFor="let msg of viewService.messages; let i = index" data-target="#carousel"
-            attr.data-slide-to="{{ i + 1 }}" (click)="showMessage(msg.id);">
-          <a href="javascript: void(0)">{{ msg.fromName }}</a>
-        </li>
-      </ul>
-    </div>
-  </div>
+<div class="hamburger vertical-center shadow" (click)="navOpen(true)">
+  <img src="../../../../assets/images/icons/chevron-right.svg">
```
Comment (ricetech) [inline src/app/pages/viewing/responses-list/responses-list.component.html:2]: Missing required alt attribute
```suggestion
  <img src="../../../../assets/images/icons/chevron-right.svg" alt="Open list of messages">
```

File: src/app/pages/viewing/messages/messages.component.html [hunk 5]
```diff
@@ -112,3 +172,17 @@ <h5 class="modal-title" id="delete-message-modal-title">Easy now!</h5>
...
+    <strong class="mr-auto">Bootstrap</strong>
+    <small>11 mins ago</small>
+    <button type="button" class="ml-2 mb-1 close" data-dismiss="toast" aria-label="Close">
+      <span aria-hidden="true">&times;</span>
+    </button>
+  </div>
+  <div class="toast-body">
+    Hello, world! This is a toast message.
+  </div>
+</div>
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.html:188]: Why is this here

File: src/app/pages/viewing/messages/messages.component.html [hunk 6]
```diff
@@ -8,36 +8,59 @@
...
+              </td>
+              <td>
+                <table>
+                  <tr>
+                    <div class="msg-header-text">
+                      <span>{{ msg.fromName }}</span>
+                    </div>
+                  </tr>
+                  <tr>
+                    <span class="msg-header-date">{{ viewService.timeSince(viewService.nowMillis, msg.timestamp.toMillis(), msg.timestamp.toDate().toDateString()) }}</span>
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.html:36]: There are 6 cases of "element div/span is not allowed here" throughout this file. Please run inspect code on your files.

File: src/app/pages/viewing/responses-list/responses-list.component.html [hunk 7]
```diff
@@ -1,70 +1,32 @@
...
+       attr.data-slide-to="{{ i + 1 }}" (click)="showMessage(msg.id);">
+    <table class="header-table">
+      <tr>
+        <td class="profile-pic-col">
+          <img class="profile-pic" src="{{ msg.fromPic }}" alt="profile-pic">
+        </td>
+        <td class="author-col">
+          <table>
+            <tr>
+              <div class="msg-header-text">
```
Comment (ricetech) [inline src/app/pages/viewing/responses-list/responses-list.component.html:20]: Please run code inspection - div isn't allowed here

File: src/app/pages/viewing/responses-list/responses-list.component.html [hunk 8]
```diff
@@ -1,70 +1,32 @@
...
+                <a href="javascript: void(0)" class="author-name">{{ msg.fromName }}</a>
+              </div>
+            </tr>
+            <tr>
+              <span class="msg-header-date">{{ viewService.timeSince(viewService.nowMillis, msg.timestamp.toMillis(), msg.timestamp.toDate().toDateString()) }}</span>
```
Comment (ricetech) [inline src/app/pages/viewing/responses-list/responses-list.component.html:25]: Please run code inspection - span isn't allowed here

File: src/app/pages/viewing/messages/messages.component.ts [hunk 9]
```diff
@@ -13,11 +16,50 @@ import { Fonts } from '../../../shared/interfaces';
...
     this.fonts = FontService.fonts;
+    this.viewService.nowMillis = Timestamp.now().toMillis();
+  }
+
+  /**
+   * Generate the lightness value of HSL from RBG
+   */
+  getLightness(r, g, b) {
+    r /= 255, g /= 255, b /= 255;
+    const max = Math.max(r, g, b), min = Math.min(r, g, b);
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.ts:31]: Multiple variable declarations per line are not permitted
```suggestion
    const max = Math.max(r, g, b);
    const min = Math.min(r, g, b);
```

File: src/app/pages/viewing/messages/messages.component.ts [hunk 10]
```diff
@@ -13,11 +16,50 @@ import { Fonts } from '../../../shared/interfaces';
...
   ngOnInit(): void {
     this.fonts = FontService.fonts;
+    this.viewService.nowMillis = Timestamp.now().toMillis();
+  }
+
+  /**
+   * Generate the lightness value of HSL from RBG
+   */
+  getLightness(r, g, b) {
+    r /= 255, g /= 255, b /= 255;
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.ts:30]: "Reports comma expressions. Comma expressions are often a sign of overly clever code, and may lead to subtle bugs. Comma expressions in the initializer or update section of a for loop are ignored by this inspection."
```suggestion
    r /= 255;
    g /= 255;
    b /= 255;
```

File: src/app/pages/viewing/messages/messages.component.ts [hunk 11]
```diff
@@ -13,11 +16,50 @@ import { Fonts } from '../../../shared/interfaces';
...
+    const max = Math.max(r, g, b), min = Math.min(r, g, b);
+    const l = (max + min) / 2;
+    return l;
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.ts:33]: Redundant (and poorly named) variable
```suggestion
    return (max + min) / 2;
```

File: src/app/pages/viewing/viewing.service.ts [hunk 12]
```diff
@@ -1,6 +1,8 @@
 import { Injectable } from '@angular/core';
 import { MsgIoService } from '../../core/msg-io.service';
 import { VirtrolioMessage } from '../../shared/interfaces';
+import { firestore } from 'firebase/app';
+import Timestamp = firestore.Timestamp;
```
Comment (ricetech) [inline src/app/pages/viewing/viewing.service.ts:5]: Unused import statement
```suggestion
```

File: src/app/pages/viewing/messages/messages.component.ts [hunk 13]
```diff
@@ -34,9 +76,9 @@ export class MessagesComponent implements OnInit {
   deleteMessage() {
     // noinspection JSIgnoredPromiseFromCall
     this.viewService.msgIo.deleteMessage(this.messageToDelete).then(() => {
-      // TODO: Add a toast on successful deletion
+      this.toastr.success('Message deleted successfully', 'Poof!');
     }).catch(e => {
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.ts:80]: Not sure how I feel about this because of the fact that the toast disappears on its own makes error reporting more difficult.
```suggestion
    }).catch(e => {
      AuthService.displayError(e);
```

File: src/app/pages/viewing/responses-list/responses-list.component.ts [hunk 14]
```diff
@@ -13,7 +13,7 @@ export class ResponsesListComponent implements OnInit {
   public displayName: string;
   public photoUrl: string;
   public uid: string;
-
+  navIsOpen = false;
   constructor(public viewService: ViewingService, private vps: ViewportScroller, public authService: AuthService,
               private route: ActivatedRoute, private router: Router) {
```
Comment (ricetech) [inline src/app/pages/viewing/responses-list/responses-list.component.ts:18]: Unused imports
```suggestion
  constructor(public viewService: ViewingService, private vps: ViewportScroller, public authService: AuthService) {
```

File: src/app/pages/viewing/messages/messages.component.html [hunk 15]
```diff
@@ -49,22 +72,59 @@
...
+                <tr>
+                  <td>
+                    <span class="profile-pic-container-small mr-2">
+                      <img class="profile-pic" src="../../../../assets/images/logo_reg.png" alt="profile-pic">
+                    </span>
+                  </td>
+                  <td>
+                    <table>
+                      <tr>
+                        <span class="msg-header-text">Vitrolio Team</span>
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.html:90]: BOI
```suggestion
                        <span class="msg-header-text">Virtrolio Team</span>
```

File: src/app/pages/viewing/responses-list/responses-list.component.css [hunk 16]
```diff
@@ -1,24 +1,106 @@
...
+  overflow-x: hidden;
+  transition: 0.5s;
+  padding-top: 2em;
 }
 
-.response-list-card-relative {
-  background-color: transparent;
-  position: relative;
-  width: auto;
+.sidenav .closebtn {
```
Comment (ricetech) [inline src/app/pages/viewing/responses-list/responses-list.component.css:15]: Typo
```suggestion
.sidenav .close-btn {
```

File: src/app/pages/viewing/responses-list/responses-list.component.css [hunk 17]
```diff
@@ -1,24 +1,106 @@
...
+  top: 0;
+  right: 25px;
+  font-size: 36px;
+  margin-left: 50px;
 }
 
-#author-list {
-  height: 10em;
-  overflow-y: scroll;
+.closebtn:hover {
```
Comment (ricetech) [inline src/app/pages/viewing/responses-list/responses-list.component.css:24]: Typo
```suggestion
.close-btn:hover {
```

File: src/app/pages/viewing/responses-list/responses-list.component.html [hunk 18]
```diff
@@ -1,70 +1,32 @@
...
-          <ul class="nav flex-column ml-2">
-            <li class="nav-item mb-1" *ngFor="let msg of viewService.messages" (click)="showMessage(msg.id);">
-              <a href="javascript: void(0)">{{ msg.fromName }}</a>
-            </li>
-          </ul>
-        </div>
-      </div>
-    </div>
+<div id="mySidenav" class="sidenav shadow" [class.navOpen]="navIsOpen">
+  <a href="javascript:void(0)" class="closebtn" (click)="navOpen(false)">&times;</a>
```
Comment (ricetech) [inline src/app/pages/viewing/responses-list/responses-list.component.html:6]: Typo
```suggestion
  <a href="javascript:void(0)" class="close-btn" (click)="navOpen(false)">&times;</a>
```

File: src/app/core/footer/footer.component.css [hunk 19]
```diff
@@ -9,6 +9,7 @@ footer {
   height: var(--footer-height);
   display: flex;
   flex-direction: column;
+  z-index: 1000;
```
Comment (ricetech) [inline src/app/core/footer/footer.component.css:12]: ...why 1000? Isn't the point to make it not selectable?

File: src/app/pages/viewing/messages/messages.component.css [hunk 20]
```diff
@@ -20,71 +9,94 @@
...
   font-size: 2em;
   color: var(--light-grey);
 }
 
-.delete-button {
+/* Remove the blue outline around the button when it's clicked */
+/*.btn:focus, .btn:active {*/
+/*  outline: none !important;*/
+/*  box-shadow: none !important;*/
+/*}*/
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.css:86]: If you don't plan on using this code, you should remove it
```suggestion
```

File: src/app/pages/viewing/messages/messages.component.html [hunk 21]
```diff
@@ -112,3 +172,17 @@ <h5 class="modal-title" id="delete-message-modal-title">Easy now!</h5>
     </div>
   </div>
 </div>
+
+<div class="toast" role="alert" aria-live="assertive" aria-atomic="true">
+  <div class="toast-header">
+    <img src="..." class="rounded mr-2" alt="...">
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.html:178]: This image is throwing a 404 error because the file `...` doesn't exist.

File: src/app/pages/viewing/messages/messages.component.ts [hunk 22]
```diff
@@ -13,11 +16,50 @@ import { Fonts } from '../../../shared/interfaces';
...
+  generateHeaderColor(backColor: string) {
+    const bgR = parseInt(backColor.slice(1, 3), 16);
+    const bgG = parseInt(backColor.slice(3, 5), 16);
+    const bgB = parseInt(backColor.slice(5), 16);
+    const hR = bgR + 20 > 255 ? 255 : bgR + 20;
+    const hG = bgG + 20 > 255 ? 255 : bgG + 20;
+    const hB = bgB + 20 > 255 ? 255 : bgB + 20;
+    const headerColor = hR.toString(16) + hG.toString(16) + hB.toString(16);
+    let headerTextColor;
+    if (this.getLightness(hR, hG, hB) > 0.65) {
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.ts:49]: You can play with the value, but right now it's too large and is rendering text unreadable:
image
image
```suggestion
    if (this.getLightness(hR, hG, hB) > 0.5) {
```

File: src/app/pages/viewing/messages/messages.component.ts [hunk 23]
```diff
@@ -13,11 +16,50 @@ import { Fonts } from '../../../shared/interfaces';
...
+      headerTextColor = '#FFFFFF';
+    }
+    return { bg: '#' + headerColor, text: headerTextColor };
+  }
+
+  checkFont(font: string) {
+    if (font in FontService.fonts) {
+      return font;
+    } else {
+      return 'Arial';
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.ts:61]: Missing backup font
```suggestion
      return 'Arial, sans-serif';
```

File: src/app/pages/viewing/messages/messages.component.ts [hunk 24]
```diff
@@ -34,9 +76,9 @@ export class MessagesComponent implements OnInit {
   deleteMessage() {
     // noinspection JSIgnoredPromiseFromCall
     this.viewService.msgIo.deleteMessage(this.messageToDelete).then(() => {
-      // TODO: Add a toast on successful deletion
+      this.toastr.success('Message deleted successfully', 'Poof!');
```
Comment (ricetech) [inline src/app/pages/viewing/messages/messages.component.ts:79]: I don't know if these are configurable, but I don't think having the toast in the top right corner is a good idea - especially since it obscures the navbar and isn't centred on mobile.
Maybe put them at the bottom of the screen like the cookie message?

File: src/app/pages/viewing/responses-list/responses-list.component.ts [hunk 25]
```diff
@@ -34,23 +34,16 @@ export class ResponsesListComponent implements OnInit {
...
-        messageId: id
-      },
-      // Preserve the existing query params in the
-      queryParamsHandling: 'merge',
-      // Update the URL
-      skipLocationChange: false
-    });
-    if (this.viewService.isCarouselView) {
+    if (!this.viewService.isCarouselView) {
+      console.log(id);
```
Comment (ricetech) [inline src/app/pages/viewing/responses-list/responses-list.component.ts:46]: Uhh... why is there a `console.log()` in here?
```suggestion
```

File: src/app/pages/viewing/viewing.service.ts [hunk 26]
```diff
@@ -18,4 +21,30 @@ export class ViewingService {
...
+    }
+    if (secondsPast < 86400) {
+      return (Math.round(secondsPast / 3600)).toString() + 'h ago';
+    }
+    if (secondsPast <= 604800) {
+      return (Math.round(secondsPast / 86400)).toString() + ' days ago';
+    }
+    if (secondsPast > 604800) {
+      // Return the full date if the Timestamp is past a week ago
+      return date;
```
Comment (ricetech) [inline src/app/pages/viewing/viewing.service.ts:47]:
```suggestion
    if (secondsPast < 60) {
      return Math.round(secondsPast).toString() + 's ago';
    } else if (secondsPast < 3600) {
      return (Math.round(secondsPast / 60)).toString() + 'm ago';
    } else if (secondsPast < 86400) {
      return (Math.round(secondsPast / 3600)).toString() + 'h ago';
    } else if (secondsPast <= 604800) {
      return (Math.round(secondsPast / 86400)).toString() + ' days ago';
    } else if (secondsPast > 604800) {
      // Return the full date if the Timestamp is past a week ago
      return date;
```

File: src/app/core/footer/footer.component.css (hunk 19 above)
Comment (GuyInFridge) [reply src/app/core/footer/footer.component.css:12]: Lol why does the footer need a `z-index` at all?

**Comment by GuyInFridge** [comment]
@janakitti is there any way to make the svg's also change colour? Cause I ended up with grey trash icons on a bright red background which didn't look good. Maybe just give them a black/white background that toggles based on the message colour?

File: src/app/core/footer/footer.component.css (hunk 19 above)
Comment (janakitti) [reply src/app/core/footer/footer.component.css:12]: We need the footer to be on top of all other elements on the site
